embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/EmbraceImpl.java

```diff
@@ -623,6 +623,12 @@ private void startImpl(@NonNull Context context,
         backgroundActivityService = sessionModule.getBackgroundActivityService();
         serviceRegistry.registerServices(sessionService, backgroundActivityService);
 
+        // Send any sessions that were cached and not yet sent.
```

Comment:
This makes a lot of sense. The triggerer of the sending of cached sessions should be done by separate component or maybe even my the delivery service itself, as you said.